src/mem/alloc.h

```diff
@@ -1069,7 +1079,7 @@ namespace snmalloc
         {
           large_allocator.memory_provider.zero(p, sizeclass_to_size(sizeclass));
         }
-        return p;
+        return Pal::apply_bounds(p, size);
```

Comment:
Can't the freelists contain bounded pointers?  Or do you want a more precise bound than the allocation size, i.e. the request size? Personally, I would prefer the free lists to contain bounded pointers.  Then we can apply the bounds at deallocation based on the pagemap lookup, and that is the only place it is required.

Reply:
I'd be content bounding things to the underlying allocation size, I think.  And I don't immediately see any reason the free lists couldn't store bounded pointers; I'll give that a shot.